server/src/modules/admin/admin.service.ts

```diff
@@ -193,4 +193,21 @@ export class AdminService {
         const dbResponse = await this.userRepo.getAllReports(where, skip, take);
         return new PaginatedResponseDto(ReportDto, dbResponse);
     }
+
+    async updateReport(userID: number, reportID: number, reportDto: UpdateReportDto) {
+        const reports = await this.userRepo.getAllReports({ id: reportID });
+
+        if (reports[1] === 0) throw new NotFoundException('Report not found');
```

Comment:
Messy to do a `getAll` here when using the report ID, which is the primary key so guaranteed to be unique. The correct way to do this is to rewrite a `userRepo.getReport` method which uses a prisma `findUnique`, where the ID is of course a valid WhereUnique.